client/dashboard/app/hooks/site-logs-data.ts

```diff
@@ -0,0 +1,76 @@
+import { useQuery } from '@tanstack/react-query';
+import { LogType, PHPLog, ServerLog, FilterType } from '../../data/site-logs';
+import { siteLogsQuery } from '../queries/site-logs';
+import type { View } from '@wordpress/dataviews';
+
+const EMPTY_ARRAY: ( ServerLog | PHPLog )[] = [];
+
+const hasLogTypeChanged = ( logType: LogType, data: { logs?: ( PHPLog | ServerLog )[] } ) => {
+	if (
+		logType === 'php' &&
+		!! data.logs &&
+		data.logs.length > 0 &&
+		'severity' in data.logs[ 0 ] &&
+		'message' in data.logs[ 0 ]
+	) {
+		return false;
+	}
+	if (
+		logType === 'server' &&
+		!! data.logs &&
+		data.logs.length > 0 &&
+		'request_type' in data.logs[ 0 ] &&
+		'status' in data.logs[ 0 ]
+	) {
+		return false;
+	}
+	return true;
+};
+
+export function useSiteLogsData( {
+	siteId,
+	view,
+	logType,
+	startTime,
+	endTime,
+	filter,
+}: {
+	siteId: number;
+	view: View;
+	logType: LogType;
+	startTime: number;
+	endTime: number;
+	filter: FilterType;
+} ) {
+	const params: SiteLogsParams = {
+		logType,
+		start: startTime,
+		end: endTime,
+		filter,
+		sortOrder: view.sort?.direction,
+		pageSize: view.perPage,
+		pageIndex: view.page,
+	};
+
+	const { data: siteLogs, isLoading, isFetching } = useQuery( siteLogsQuery( siteId, params ) );
+
+	const logs =
+		! siteLogs?.logs || hasLogTypeChanged( logType, siteLogs )
```

Comment:
I'm a bit confused about this `hasLogTypeChanged()` logic. Will it be possible / is it necessary at all? Because when `logType` changes, so will the result of `useQuery( siteLogsQuery( siteId, params ) );`, so the `siteLogs` will be updated accordingly (either `undefined` when still loading, or the logs with the correct type).

Reply:
Good point. That was copied over initially, but no it doesn't appear to be needed. Removed now.